pdata/pprofile/mappings.go

```diff
@@ -0,0 +1,47 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pprofile // import "go.opentelemetry.io/collector/pdata/pprofile"
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+var errTooManyMappingTableEntries = errors.New("too many entries in MappingTable")
+
+// SetMapping updates a MappingTable and a Location's MappingIndex to
+// add or update a mapping.
+func SetMapping(table MappingSlice, record Location, ma Mapping) error {
+	idx := int(record.MappingIndex())
+	if idx > 0 {
+		if idx >= table.Len() {
+			return fmt.Errorf("index value %d out of range for MappingIndex", idx)
+		}
+		mapAt := table.At(idx)
+		if mapAt.Equal(ma) {
+			// Mapping already exists, nothing to do.
+			return nil
+		}
+	}
+
+	for j, a := range table.All() {
+		if a.Equal(ma) {
```

Comment:
Maybe use `m` for map instead of `a` for `attribute`?
```suggestion
	for j, m := range table.All() {
		if m.Equal(ma) {
```